scripts-available/CDB_CartodbfyTable.sql

```diff
@@ -649,3 +649,837 @@ BEGIN
+        || ', 3857) AS '
+        || const.mercgeomcol,
+        a.attname
+      INTO geom_transform_sql, geom_column_source
+      FROM pg_class c 
+      JOIN pg_attribute a ON a.attrelid = c.oid 
+      JOIN pg_type t ON a.atttypid = t.oid,
+      ( SELECT rec.srid AS srid ) AS srid
+      -- ( SELECT table_srid AS srid ) AS srid
+      WHERE c.oid = reloid
+      AND t.typname = 'geometry'
+      AND a.attnum > 0
+      AND NOT a.attisdropped
+      AND (postgis_typmod_srid(a.atttypmod) > 0 OR srid.srid > 0)
+      ORDER BY a.attnum
+      LIMIT 1;
+    
+      IF FOUND THEN
+        sql := sql || geom_transform_sql;
+      END IF;
+
+    END IF;
+    
+  END IF;
+
+  -- Add now add all the rest of the columns
+  -- by selecting their names into an array and
+  -- joining the array with a comma
+  SELECT 
+    ',' || array_to_string(array_agg(a.attname),',') AS column_name_sql, 
+    Count(*) AS count
+  INTO rec
+  FROM pg_class c 
+  JOIN pg_attribute a ON a.attrelid = c.oid 
+  JOIN pg_type t ON a.atttypid = t.oid
+  WHERE c.oid = reloid
+  AND a.attnum > 0
+  AND a.attname NOT IN (const.geomcol, const.mercgeomcol, const.pkey, geom_column_source)
+  AND NOT a.attisdropped;
+
+
+  -- No non-cartodb columns? Possible, I guess.
+  IF rec.count = 0 THEN
+    RAISE DEBUG 'CDB(_CDB_Rewrite_Table): %', 'found no extra columns';
+    column_name_sql := '';
+  ELSE
+    RAISE DEBUG 'CDB(_CDB_Rewrite_Table): %', Format('found extra columns columns ''%s''', rec.column_name_sql);
+    column_name_sql := rec.column_name_sql;
+  END IF;
+
+  -- Add the source table to the SQL
+  sql := sql || column_name_sql || ' FROM ' || reloid::text;
+  RAISE DEBUG 'CDB(_CDB_Rewrite_Table): %', sql;
+
+  -- Run it!
+  PERFORM _CDB_SQL(sql, '_CDB_Rewrite_Table');
+  
+  -- Set up the primary key sequence
+  -- If we copied the primary key from the original data, we need
+  -- to set the sequence to the maximum value of that key
+  IF has_usable_primary_key THEN
+
+    EXECUTE Format('SELECT max(%s) FROM %s',
+            const.pkey, copyname)
+       INTO destseqmax;
+
+    IF FOUND AND destseqmax IS NOT NULL THEN
+      PERFORM _CDB_SQL(Format('SELECT setval(''%s'', %s)', destseq, destseqmax), '_CDB_Rewrite_Table');
+    END IF;
+
+  END IF;
+
+  -- Make the primary key use the sequence as its default value
+  sql := Format('ALTER TABLE %s ALTER COLUMN %I SET DEFAULT nextval(''%s'')', 
+          copyname, const.pkey, destseq);
+  PERFORM _CDB_SQL(sql, '_CDB_Rewrite_Table');
+
+  -- Make the sequence owned by the table, so when the table drops, 
+  -- the sequence does too
+  sql := Format('ALTER SEQUENCE %s OWNED BY %s.%s', destseq, copyname, const.pkey);
+  PERFORM _CDB_SQL(sql,'_CDB_Rewrite_Table');
+  
+  -- We just made a copy, so we can drop the original now
+  sql := Format('DROP TABLE %s', reloid::text);
+  PERFORM _CDB_SQL(sql, '_CDB_Rewrite_Table');
+  
+  -- If we used a temporary destination table
+  -- we can now rename it into place
```

Comment:
The re-write will have created the new table right into the destination location we wanted.  The case of the no-write-required will have already returned higher up in the function, so at this point we know we've re-written the table. Question is, did we rewrite it to a temp location or a permanent location? if the latter, then we've nothing left to do. By the time this function returns, we guarantee there's a good table in the right place, so no responsibility handed to caller.